RELEASE.md

```diff
@@ -75,24 +75,26 @@ you can skip the dependency update or only update select dependencies. In such a
 case, you have to create an issue or pull request in the GitHub project for
 later follow-up.
 
-### Prepare your release
+### 1. Prepare your release
 
-For a patch release, work in the branch of the minor release you want to patch.
+For a new major or minor release, create the corresponding release branch based on the master branch. Note that a new a new release branch in `origin` is only necessary if there is not already one for this <major>.<minor> version. For example if we're releasing 2.17.0 and the previous stable release is 2.16.0 we need to create a `release-2.17` branch. Patch release don't require their own `release-<major>.<minor>.<patch>, and the same for release candidate versions.
```

Comment:
If you want to go into this level of detail, I'd say you should use a stricter wording. It's not about not requiring other branches, you _must not_ create any other branches starting with `release-...`. (This naming pattern is used to mark those branches as protected. It is very tedious to remove a `release-2.16.0` branch.)

The example is a bit misleading because the usual scenario is that `release-2.17` is created just before cutting 2.17.0-rc.0 rather than 2.17.0.

Suggestion for the whole paragraph:

"""
All releases are handled in protected release branches. They follow the naming scheme `release-<major>.<minor>`. (Branch protection kicks in automatically for any branches whose name starts with `release-`. Never use names starting with `release-` for branches that are not release branches.) Release candidates and patch releases for any given major or minor release happen in the same `release-<major>.<minor>` branch. If the relevant `release-<major>.<minor>` branch for your release doesn't already exist (which is usually the case while cutting the first release candidate for a new major or minor release), create it based on the master branch.
"""

Reply:
PTAL. I added your suggestions regarding the branch protection details to the earlier section `branch management and versioning`, and added a reference to that section in this portion of the doc.

I wanted this section to be detailed because so far most of what's come up in this PR (branch protection, not creating rc/patch branches regardless of the actual name of the branch, etc.) was not clear to me from reading the previous doc.